### ISSUE TITLE
Sequential mail sending aborts after mail is rejected by mailserver.

### ISSUE THREAD

**Issue by rFranzr** [SAP/cloud-sdk-js#5172]
**Describe the bug**
Sequential mail sending aborts after one mail is rejected by the mailserver.

**To Reproduce**

We use cloud-sdk to send emails sequentially. 
Given we have got two addresses: `['user.non-exisiting@sap.com', 'user.existing@sap.com']`

When the mailserver rejects an email, the following emails won't be sent - sending seems to be aborted.

1. Send Emails to both recipients sequentially
2. First recipient is rejected
```Can't send mail - all recipients were rejected: 550 5.1.1 <user.non-existing@sap.com>: Recipient address rejected: Address not found.```
5. Second mail is not delivered

*** Mail Client Options ***
```
const mailerOptions: MailClientOptions = {
  'requireTLS': true,
  'sdkOptions': {
    parallel: false
  },
  'tls': {
    'rejectUnauthorized': false
  }
}
```

**Expected behavior**
We expect all valid emails to be delivered.

**Used Versions:**

- node version 20.15.1
- npm version 10.7.0
- SAP Cloud SDK version 3.22.2

**Impact / Priority**
Affected development phase: e.g. Production

Impact: Impaired

Timeline: already-live.

**Comment by jjtang1985** [comment]
Hi @rFranzr,

Thanks for approaching.

Have you tried to send them in parallel?
Could you please share your code snippets?
What will happen, if you just send the second mail? Will it be delivered?

**Comment by rFranzr** [comment]
Hi @jjtang1985, 

Sending emails in parallel leads to the same result.
Sending to just one valid recipient works.

`
const destinationOption: DestinationFetchOptions = {
  'destinationName': 'smtp'
}

export const sendMails = async (mailConfigs: MailConfig[]): Promise<void> => {
  const validatedMails = mailConfigs.filter(validateMailRecipients)

  if (validatedMails.length === 0) {
    LOG.warn('No valid Mails to send.')
    return
  }
  LOG.info('Sending %s mails.', [validatedMails.length])
  const mailResponses = await sendMail(
    destinationOption,
    validatedMails,
    mailerOptions
  )

  await handleMailSendingResponses(mailResponses)
}
`

**Comment by deekshas8** [comment]
Hi @rFranzr , I created a canary version if you want to try it out. We'll release a new version tomorrow.

**Comment by deekshas8** [comment]
The fix is now available in the latest v3.23.0. 
I'll close the issue, please reopen if you need further support.

**Comment by rFranzr** [comment]
@deekshas8 Thanks for the great support and quick fix!
